File: app/src/main/java/com/yatatsu/passiveviewsample/ui/input/InputController.java
package com.yatatsu.passiveviewsample.ui.input;


import com.yatatsu.passiveviewsample.ui.base.Controller;

public class InputController extends Controller<InputScreen> {

    void onSubmitEditText(String input) {
        if (input == null || input.isEmpty()) {
            getScreen().showError("input must not be empty.");
            return;
        }
        input = input.trim();
        if (input.contains(" ")) {
            getScreen().showError("should be one word.");
        } else {
            getScreen().hideError();
            getScreen().navigateToUserScreen(input);
        }
    }
}

File: app/src/main/java/com/yatatsu/passiveviewsample/ui/input/InputController.kt
package com.yatatsu.passiveviewsample.ui.input


import com.yatatsu.passiveviewsample.ui.base.Controller

class InputController : Controller<InputScreen>() {

  fun onSubmitEditText(input: String?) {
    if (input == null || input.isEmpty()) {
      screen?.showError("input must not be empty.")
      return
    }
    val trimmed = input.trim { it <= ' ' }
    if (trimmed.contains(" ")) {
      screen?.showError("should be one word.")
    } else {
      screen?.hideError()
      screen?.navigateToUserScreen(trimmed)
    }
  }
}
